src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -575,14 +578,29 @@ public void Dispose()
                 await ReadCharAsync(cancellationToken).ConfigureAwait(false) != 'T' ||
                 await ReadCharAsync(cancellationToken).ConfigureAwait(false) != 'T' ||
                 await ReadCharAsync(cancellationToken).ConfigureAwait(false) != 'P' ||
-                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '/' ||
-                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '1' ||
-                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.' ||
-                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '1')
+                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '/')
             {
                 throw new HttpRequestException("could not read response HTTP version");
             }
 
+            // Set the response HttpVersion.
+            char majorVersion = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+            if (!char.IsDigit(majorVersion) ||
+                await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.')
+            {
+                throw new HttpRequestException("could not read response HTTP version");
+            }
+            char minorVersion = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+            if (!char.IsDigit(minorVersion))
+            {
+                throw new HttpRequestException("could not read response HTTP version");
+            }
+            response.Version =
+                (majorVersion == '1' && minorVersion == '1') ? HttpVersionInternal.Version11 :
+                (majorVersion == '1' && minorVersion == '0') ? HttpVersionInternal.Version10 :
+                (majorVersion == '2' && minorVersion == '0') ? HttpVersionInternal.Version20 :
```

Comment:
I don't think we should ever see 2.0 here.  2.0 will not be seen unless you do a connection upgrade.  That said, I guess it doesn't hurt to check for it.